spec/models/reservation_spec.rb

```diff
@@ -664,6 +664,46 @@
     expect(reservation).to be_can_start_early
   end
 
+  describe '#start_reservation!' do
+    it 'sets actual start time', :timecop_freeze do
+      reservation.start_reservation!
+      expect(reservation.actual_start_at).to eq(Time.now)
+    end
+
+    context 'with a running reservation' do
+      let!(:running) { create :setup_reservation, product: instrument, reserve_start_at: (Time.now - 1.hour), reserve_end_at: Time.now, actual_start_at: (Time.now - 1.hour) }
+      before do
+        order = running.order_detail.order
+        order.state = 'validated'
+        order.purchase!
+      end
+
+      it 'completes the running reservation' do
+        reservation.start_reservation!
+        expect(running.reload).to be_complete
+      end
+
+      it 'sets the orders as a problem order' do
```

Comment:
I would probably add a test that it doesn't set the actual_end_at of the previously-running reservation.